src/analysis/retail/evoker/preservation/modules/talents/AncientFlame.tsx

```diff
@@ -0,0 +1,150 @@
+import { formatDuration } from 'common/format';
+import SPELLS from 'common/SPELLS';
+import { TALENTS_EVOKER } from 'common/TALENTS';
+import { SpellLink } from 'interface';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { ApplyBuffEvent, RefreshBuffEvent, RemoveBuffEvent } from 'parser/core/Events';
+import { BoxRowEntry, PerformanceBoxRow } from 'interface/guide/components/PerformanceBoxRow';
+import { QualitativePerformance } from 'parser/ui/QualitativePerformance';
+import { RoundedPanel } from 'interface/guide/components/GuideDivs';
+import { reduce } from 'vega-lite/build/src/encoding';
+import { GUIDE_CORE_EXPLANATION_PERCENT } from '../../Guide';
+import { explanationAndDataSubsection } from 'interface/guide/components/ExplanationRow';
+
+class AncientFlame extends Analyzer {
+  consumptions: BoxRowEntry[] = [];
+  lastApply: ApplyBuffEvent | null = null;
+  consumeTimes: number[] = [];
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS_EVOKER.ANCIENT_FLAME_TALENT);
+    this.addEventListener(
+      Events.refreshbuff.by(SELECTED_PLAYER).spell(SPELLS.ANCIENT_FLAME_BUFF),
+      this.onRefresh,
+    );
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.ANCIENT_FLAME_BUFF),
+      this.onApply,
+    );
+    this.addEventListener(
+      Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.ANCIENT_FLAME_BUFF),
+      this.onConsume,
+    );
+  }
+
+  onApply(event: ApplyBuffEvent) {
+    this.lastApply = event;
+  }
+
+  get applyTime() {
+    return this.lastApply?.timestamp || this.owner.fight.start_time;
+  }
+
+  onRefresh(event: RefreshBuffEvent) {
+    let tooltip = null;
+    let value = QualitativePerformance.Good;
+    // 1 stack after cast = they were at 2 so not a waste
+    if (this.selectedCombatant.getBuffStacks(SPELLS.ESSENCE_BURST_BUFF.id) === 1) {
```

Comment:
Probably need to track last proccing cast, this misses the refresh event triggered by verdant embrace w/1 EB stack which would be a bad cast. 

1 Stack Essence Burst & Ancient Flame Active  -> VE Cast -> Bad Refresh

Or at the very least the tooltip would be inaccurate for this scenario